driver-embedded/src/main/com/mongodb/embedded/client/MongoClients.java

```diff
@@ -35,6 +42,26 @@
      * @param mongoEmbeddedSettings the settings for the embedded driver.
      */
     public static void init(final MongoEmbeddedSettings mongoEmbeddedSettings) {
```

Comment:
All three methods of this class should be synchronized to avoid concurrent calls to init, fini, and db_new, and also to publish the change to mongoDBCAPI global.

Reply:
Done - in a new MongoDBCAPIHelper class.